compat/v1/compat.go

```diff
@@ -0,0 +1,323 @@
+				fmt.Errorf("Failed to unmarshal digibyte shard state: %v", err)
+		}
+
+		stateResponse.Digibyte = UTXOState{
+			Address:           dgbShard.Shard.String(),
+			Dust:              digibyteS.DustAmount.String(),
+			Gascap:            digibyteS.GasCap.String(),
+			Gaslimit:          digibyteS.GasLimit.String(),
+			Gasprice:          digibyteS.GasPrice.String(),
+			Latestchainheight: digibyteS.LatestHeight.String(),
+			Minimumamount:     digibyteS.MinimumAmount.String(),
+			Output: Output{
+				Outpoint: Outpoint{
+					Hash:  dgbOutput.Hash.String(),
+					Index: dgbOutput.Index.String(),
+				},
+				Pubkeyscript: dgbOutput.PubKeyScript.String(),
+				Value:        dgbOutput.Value.String(),
+			},
+			Pubkey: dgbShard.PubKey.String(),
+		}
+
+	}
+
+	dogecoinS, ok := state[string(multichain.DigiByte.NativeAsset())]
+	if ok {
+
+		if len(dogecoinS.Shards) == 0 {
+			return StateResponse{},
+				fmt.Errorf("No Dogecoin Shards")
+		}
+
+		dogeShard := dogecoinS.Shards[0]
+
+		var dogeOutput engine.XStateShardUTXO
+		if err := pack.Decode(&dogeOutput, dogeShard.State); err != nil {
+			return StateResponse{},
+				fmt.Errorf("Failed to unmarshal dogecoin shard state: %v", err)
+		}
+
+		stateResponse.Dogecoin = UTXOState{
+			Address:           dogeShard.Shard.String(),
+			Dust:              dogecoinS.DustAmount.String(),
+			Gascap:            dogecoinS.GasCap.String(),
+			Gaslimit:          dogecoinS.GasLimit.String(),
+			Gasprice:          dogecoinS.GasPrice.String(),
+			Latestchainheight: dogecoinS.LatestHeight.String(),
+			Minimumamount:     dogecoinS.MinimumAmount.String(),
+			Output: Output{
+				Outpoint: Outpoint{
+					Hash:  dogeOutput.Hash.String(),
+					Index: dogeOutput.Index.String(),
+				},
+				Pubkeyscript: dogeOutput.PubKeyScript.String(),
+				Value:        dogeOutput.Value.String(),
+			},
+			Pubkey: dogeShard.PubKey.String(),
+		}
+
+	}
+
+	terraS, ok := state[string(multichain.Terra.NativeAsset())]
+	if ok {
+
+		if len(terraS.Shards) == 0 {
+			return StateResponse{},
+				fmt.Errorf("No Terra Shards")
+		}
+
+		lunaShard := terraS.Shards[0]
+
+		var lunaOutput engine.XStateShardAccount
+		if err := pack.Decode(&lunaOutput, lunaShard.State); err != nil {
+			return StateResponse{},
+				fmt.Errorf("Failed to unmarshal terra shard state: %v", err)
+		}
+
+		terra := AccountState{
+			Address:           lunaShard.Shard.String(),
+			Gascap:            terraS.GasCap.String(),
+			Gaslimit:          terraS.GasLimit.String(),
+			Gasprice:          terraS.GasPrice.String(),
+			Latestchainheight: terraS.LatestHeight.String(),
+			Minimumamount:     terraS.MinimumAmount.String(),
+			Nonce:             lunaOutput.Nonce.String(),
+			Pubkey:            lunaShard.PubKey.String(),
+		}
+
+		for _, v := range lunaOutput.Gnonces {
+			gnonce := Gnonces{
+				Address: v.Address.String(),
+				Nonce:   v.Nonce.String(),
+			}
+			terra.Gnonces = append(terra.Gnonces, gnonce)
+		}
+		stateResponse.Terra = terra
+
+	}
```

Comment:
I think we could probably reduce a lot of this code since the logic is the same for each chain, but I'm also not too worried about it because we shouldn't expect this code to change (if anything, it will be removed entirely once we no longer require support for earlier versions).

Reply:
Yeah, good point, could have made an "account state" builder and and a "utxo state" builder; so used to code duplication in go that I just did it manually. Am ok to leave it as thus for now

Reply:
All good